paasta_tools/cli/cmds/local_run.py

```diff
@@ -653,6 +690,82 @@ def check_if_port_free(port):
     return True
 
 
+def assume_aws_role(
+    instance_config: InstanceConfig,
+    service: str,
+    assume_role_arn: str,
+    assume_pod_identity: bool,
+    use_okta_role: bool,
+) -> AWSSessionCreds:
+    """Runs AWS cli to assume into the correct role, then extract and return the ENV variables from that session"""
+    pod_identity = instance_config.get_iam_role()
+    if assume_role_arn:
+        pod_identity = assume_role_arn
+    if assume_pod_identity and not pod_identity:
+        print(
+            f"Error: --assume-pod-identity passed but no pod identity was found for this instance ({instance_config.instance})",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    with open("/nail/etc/runtimeenv") as runtimeenv_file:
+        aws_account = runtimeenv_file.read()
+    if pod_identity and (assume_pod_identity or assume_role_arn):
+        print(
+            "Calling aws-okta to assume role {} using account {}".format(
+                pod_identity, aws_account
+            )
+        )
+    elif use_okta_role:
+        print(f"Calling aws-okta using account {aws_account}")
+    else:
+        # use_okta_role, assume_pod_identity, and assume_role are all empty. This shouldn't happen
```

Comment:
should we print something here instead of silently returning nothing? or maybe error out in this case?

Reply:
i'd probably err on the side of erroring out unless we want to break this function up into multiple functions and have the code that calls `assume_aws_role()` instead call the specific version that does what the cli args call for (e.g., `assume_okta_role()`, `assume_pod_identity_role()`, `assume_role_arn()`)